test/CheckedC/dump-dataflow-facts.c

```diff
@@ -0,0 +1,213 @@
+// Tests for dumping of datafow analysis for collecting facts
+//
+// RUN: %clang_cc1 -DDEBUG_DATAFLOW -fcheckedc-extension %s 2>&1 | FileCheck %s
+
+//===================================================================
+// Dumps of different kinds of collected dataflow facts
+//===================================================================
+
+
+int f(int a);
+
+void fn_1(void) {
+  int a, b, c;
+  if (b < c)
+    if (f((a=5)+3))
+      b = c;
+}
+
+// CHECK: {
```

Comment:
This should include block number information.  Otherwise this will be very difficult to debug in the future.

Could you add more test cases for more complicated expressions?